scripts/splitControl.js

```diff
@@ -0,0 +1,129 @@
+/*global $ */
+/* exported initSplit */
+
+function initSplit(newData) {
+    // use these defaults if any newData not set
+    let data = {paneContainer: "#window", verticalSplit: false, splitPercent: 50, minSiz0: 50, minSiz1: 50};
```

Comment:
paneContainer should be a required argument, not an optional one, so the function definition should look more like:
```
function initSplit(container, config) {
```

Reply:
I think it would be more intuitive if `verticalSplit` was `splitDirection` and one of `horizontal`/`vertical`.

Reply:
This should also take an optional ID for each pane and set it when creating the `<div>`s to allow programmatic access to them later. Actually, that may not work quite right as other JS (and possibly the PHP) will need to interact with these divs and we might get into a race condition on when the IDs were defined.

How do other splitters handle this? Do they require the `<div>`s to be defined already and just pass in the IDs?

Reply:
> paneContainer should be a required argument, not an optional one, so the function definition should look more like:
> 
> ```
> function initSplit(container, config) {
> ```

I agree.


> I think it would be more intuitive if `verticalSplit` was `splitDirection` and one of `horizontal`/`vertical`.

Yes, I'm afraid my desire to do things most efficiently from the time when memory and speed were only a tiny fraction of what they are now
`if(verticalSplit)` rather than `if(splitDirection === `vertical`)

Reply:
> This should also take an optional ID for each pane and set it when creating the `<div>`s to allow programmatic access to them later. Actually, that may not work quite right as other JS (and possibly the PHP) will need to interact with these divs and we might get into a race condition on when the IDs were defined.
> 
> How do other splitters handle this? Do they require the `<div>`s to be defined already and just pass in the IDs?

The previous version of this, (until a few days ago) needed the divs to be pre-defined and identified them as children of the container. This might be a better way to do it. It might be best to still create the splitter-bar div inside the splitter code but possibly not.
It would be easy to configure the width and colour of the splitter bar and it would be nice to have an image in it (like the three dots in your wire-frames). I'll look at doing this. And look at other splitters again.

Reply:
> The previous version of this, (until a few days ago) needed the divs to be pre-defined and identified them as children of the container. This might be a better way to do it. It might be best to still create the splitter-bar div inside the splitter code but possibly not.

I do like the idea of saying "to use this, pass in the ID of a `<div>` that contains **exactly two other immediate-children** `<div>`s, eg: for `<div id='container'><div></div><div></div></div>` pass in 'container'". Then the code will add the splitter bar between the two child divs and off we go. But this is also me thinking out loud a bit of how I would want to use it as a developer.

Reply:
changed to using two pre-made divs.